src/lib/containers/table/datasets/DatasetItemsEditor.tsx

```diff
@@ -187,24 +207,20 @@ export function DatasetItemsEditor(props: DatasetItemsEditorProps) {
         )
 
         // "X item(s) added"
-        let toastMessageTitle = `${newItems.length} Item${
+        let toastTitle = `${newItems.length} Item${
           newItems.length === 1 ? '' : 's'
         } added`
 
         // "and Y item(s) updated", only shown if there are updated items
         if (updatedItems.length > 0) {
-          toastMessageTitle += ` and ${updatedItems.length} Item${
+          toastTitle += ` and ${updatedItems.length} Item${
             updatedItems.length === 1 ? '' : 's'
           } updated`
         } else {
           // if no items were updated, title = "X items(s) added" + " to Dataset"
-          toastMessageTitle += ` to Dataset`
+          toastTitle += ` to Dataset`
         }
```

Comment:
Just to clean up the code a bit, can this be pulled out into more straightforward functions, each with a single responsibility (and could also be tested individually)?

In my mind you could have 
* A function that takes two lists of dataset items and returns the counts of added, changed, and removed items
* A function that takes the counts of added, changed, and removed items, and returns the appropriate toast message

Reply:
With this change, the toast message shouldn't have to be stored in state.